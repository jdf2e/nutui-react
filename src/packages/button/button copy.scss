.nut-button {
  position: relative;
  display: flex;
  display: inline-block;
  width: auto;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  height: $button-default-height;
  font-size: $button-default-font-size;
  font-weight: $button-default-font-weight;
  text-align: center;
  cursor: pointer;
  transition: $button-transition;
  -webkit-appearance: none;
  user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  color: $button-default-color;
  background: $button-default-background-color;

  &-text {
    margin-left: $button-text-icon-margin;
    &.right {
      margin-right: $button-text-icon-margin;
    }
  }

  &-children {
    display: flex;
    flex-direction: row;
  }

  &::before {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 100%;
    height: 100%;
    background-color: $color-mask;
    border: inherit;
    border-color: $color-mask;
    border-radius: inherit;
    transform: translate(-50%, -50%);
    opacity: 0;
    content: ' ';
  }

  &::after {
    border: none;
  }

  &:active::before {
    opacity: 0.1;
  }

  &-wrap {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    /* #ifndef rn harmony */
    background: initial;
    /* #endif */

    .nut-icon {
      font-size: $button-default-font-size;
      width: $button-default-font-size;
      height: $button-default-font-size;
    }
  }

  &-loading,
  &-disabled {
    &::before {
      display: none;
    }
  }

  &.nut-button-icononly {
    width: $button-default-height;
    padding: 0;
  }

  &-default {
    padding: $button-default-padding;
    background: transparent;
    border-width: $button-border-width;
    border-style: solid;
    border-color: $button-default-border-color;

    &-disabled {
      color: $button-default-disabled-color;
      border-color: $button-default-disabled;
    }

    &-none {
      border-color: transparent;
    }

    &-dashed {
      border-style: dashed;
    }

    &-outline,
    &-dashed {
      background: transparent;
    }

    &-solid-disabled,
    &-outline-disabled,
    &-dashed-disabled {
      // border-color: $button-default-disabled;
    }
  }

  &-normal {
    padding: $button-normal-padding;
  }

  &-xlarge {
    height: $button-xlarge-height;
    padding: $button-xlarge-padding;
    font-size: $button-xlarge-font-size;
    font-weight: $button-large-font-weight;

    .nut-icon {
      font-size: $button-xlarge-font-size;
      width: $button-xlarge-font-size;
      height: $button-xlarge-font-size;
    }
  }

  &-large {
    height: $button-large-height;
    padding: $button-large-padding;
    font-size: $button-large-font-size;
    font-weight: $button-large-font-weight;

    .nut-icon {
      font-size: $button-large-font-size;
      width: $button-large-font-size;
      height: $button-large-font-size;
    }
  }

  &-small {
    height: $button-small-height;
    padding: $button-small-padding;
    font-size: $button-small-font-size;

    .nut-icon {
      font-size: $button-small-font-size;
      width: $button-small-font-size;
      height: $button-small-font-size;
    }
  }

  &-mini {
    height: $button-mini-height;
    padding: $button-mini-padding;
    font-size: $button-mini-font-size;

    .nut-icon {
      font-size: $button-mini-font-size;
      width: $button-mini-font-size;
      height: $button-mini-font-size;
    }
  }

  &-primary {
    color: $button-primary-border-color;
    background-origin: border-box;
    border-width: $button-border-width;
    border-style: solid;
    border-color: $button-primary-border-color;

    &-solid {
      color: $button-primary-color;
      background: $button-primary-background-color;
      border-color: transparent;
    }

    &-dashed {
      border-style: dashed;
    }

    &-none {
      border-color: transparent;
    }

    // &-none-children, &-outline-children, &-dashed-children{
    //   // color: $button-primary-border-color;
    // }

    &-disabled {
      background: transparent;
      color: $button-primary-disabled;
      border-color: $button-primary-disabled;
    }

    &-solid-disabled {
      color: $button-primary-color;
      background: $button-primary-disabled;
    }

    &-none-disabled {
      border-color: transparent;
    }

    // &-solid-disabled-children {
    //   border-color: $button-primary-disabled;
    // }

    // &-outline-disabled-children {
    //   color: $button-primary-disabled;
    // }

    // &-dashed-disabled-children {
    //   color: $button-primary-disabled;
    // }

    // &-none-disabled-children {
    //   color: $button-primary-disabled;
    // }

    // &-children {
    //   // color: $button-primary-color;
    // }
  }

  &-success {
    color: $button-success-border-color;
    background-origin: border-box;
    border-width: $button-border-width;
    border-style: solid;
    border-color: $button-success-border-color;

    &-solid {
      color: $button-success-color;
      background: $button-success-background-color;
      border-color: transparent;
    }

    &-dashed {
      border-style: dashed;
    }

    &-none {
      border-color: transparent;
    }

    &-disabled {
      background: transparent;
      color: $button-success-disabled;
      border-color: $button-success-disabled;
    }

    &-solid-disabled {
      color: $button-success-color;
      background: $button-success-disabled;
    }

    &-none-disabled {
      border-color: transparent;
    }
  }

  &-info {
    color: $button-info-border-color;
    background: transparent;
    background-origin: border-box;
    border-width: $button-border-width;
    border-style: solid;
    border-color: $button-info-border-color;

    &-solid {
      color: $button-info-color;
      background: $button-info-background-color;
      border-color: transparent;
    }

    &-dashed {
      border-style: dashed;
    }

    &-none {
      border-color: transparent;
    }

    &-disabled {
      background: transparent;
      color: $button-info-disabled;
      border-color: $button-info-disabled;
    }

    &-solid-disabled {
      color: $button-info-color;
      background: $button-info-disabled;
    }

    &-none-disabled {
      border-color: transparent;
    }
  }

  &-danger {
    color: $button-danger-border-color;
    background-origin: border-box;
    border-width: $button-border-width;
    border-style: solid;
    border-color: $button-danger-border-color;

    &-solid {
      color: $button-danger-color;
      background: $button-danger-background-color;
      border-color: transparent;
    }

    &-dashed {
      border-style: dashed;
    }

    &-none {
      border-color: transparent;
    }

    &-disabled {
      background: transparent;
      color: $button-danger-disabled;
      border-color: $button-danger-disabled;
    }

    &-solid-disabled {
      color: $button-danger-color;
      background: $button-danger-disabled;
    }

    &-none-disabled {
      border-color: transparent;
    }
  }

  &-warning {
    color: $button-warning-border-color;
    background-origin: border-box;
    border-width: $button-border-width;
    border-style: solid;
    border-color: $button-warning-border-color;

    &-solid {
      color: $button-warning-color;
      background: $button-warning-background-color;
      border-color: transparent;
    }

    &-dashed {
      border-style: dashed;
    }

    &-none {
      border-color: transparent;
    }

    &-disabled {
      background: transparent;
      color: $button-warning-disabled;
      border-color: $button-warning-disabled;
    }

    &-solid-disabled {
      color: $button-warning-color;
      background: $button-warning-disabled;
    }

    &-none-disabled {
      border-color: transparent;
    }
  }

  &-block {
    display: block;
    width: 100%;
  }

  &-solid {
    color: $button-primary-color;
    background: $button-primary-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      color: $button-default-disabled-color;
      background: $button-default-disabled;
    }
  }

  &-outline,
  &-dashed {
    background: transparent;
  }

  &-none {
    background: transparent;
    border-color: transparent;
    background: transparent;
  }

  &-disabled {
    cursor: not-allowed;
    color: #ffffff;
    background: $button-default-disabled;
    border-color: $button-default-disabled;
  }

  &-loading {
    cursor: default;
    opacity: 0.9;
  }

  &-round {
    border-radius: $button-border-radius;

    &.nut-button-xlarge,
    &.nut-button-large {
      border-radius: $button-large-border-radius;
    }

    &.nut-button-small {
      border-radius: $button-small-border-radius;
    }

    &.nut-button-mini {
      border-radius: $button-mini-border-radius;
    }
  }

  &-square {
    border-radius: $button-square-border-radius;
  }
}

[dir='rtl'] .nut-button,
.nut-rtl .nut-button {
  &-text {
    margin-left: 0;
    margin-right: $button-text-icon-margin;

    &.right {
      margin-left: $button-text-icon-margin;
    }
  }

  &::before {
    left: auto;
    right: 50%;
    transform: translate(50%, -50%);
  }
}
