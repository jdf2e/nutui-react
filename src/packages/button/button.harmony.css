.nut-button {
  position: relative;
  display: flex;
  display: inline-block;
  /* #ifdef rn harmony*/
  width: 80px;
  /* #endif */
  /* #ifndef rn harmony*/
  width: auto;
  /* #endif */
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  height: 32px;
  cursor: pointer;
  transition: opacity 0.2s;
  -webkit-appearance: none;
  user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  background: transparent;
  border-width: 1px;
}
.nut-button-text {
  margin-left: 4px;
}
.nut-button-text-right {
  margin-right: 4px;
}
.nut-button-children {
  display: flex;
  flex-direction: row;
  background: transparent;
  font-size: 14px;
  font-weight: 400;
  text-align: center;
  color: #1a1a1a;
}
.nut-button::before {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.7);
  border: inherit;
  border-color: rgba(0, 0, 0, 0.7);
  border-radius: inherit;
  transform: translate(-50%, -50%);
  opacity: 0;
  content: " ";
}
.nut-button::after {
  border: none;
}
.nut-button:active::before {
  opacity: 0.1;
}
.nut-button-wrap {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  /* #ifndef rn harmony */
  background: initial;
  /* #endif */
}
.nut-button-wrap .nut-icon {
  font-size: 14px;
  width: 14px;
  height: 14px;
}
.nut-button-loading::before, .nut-button-disabled::before {
  display: none;
}
.nut-button-disabled {
  cursor: not-allowed;
  color: #ffffff;
}
.nut-button.nut-button-icononly {
  width: 32px;
  padding: 0;
}
.nut-button-default {
  padding: 0px 12px;
  border-style: solid;
  border-color: #505259;
}
.nut-button-default-disabled {
  color: #ffffff;
  background: #c2c4cc;
  border-color: #c2c4cc;
}
.nut-button-default-solid-disabled {
  color: #ffffff;
  background: #c2c4cc;
  border-color: #c2c4cc;
}
.nut-button-default-none-disabled {
  color: #888b94;
}
.nut-button-default-outline-disabled {
  background: transparent;
  color: #c2c4cc;
  border-color: #c2c4cc;
}
.nut-button-default-dashed-disabled {
  background: transparent;
  color: #c2c4cc;
  border-color: #c2c4cc;
}
.nut-button-normal {
  padding: 0px 12px;
}
.nut-button-xlarge {
  height: 48px;
  padding: 0px 24px;
  font-size: 24px;
  font-weight: 600;
}
.nut-button-xlarge .nut-icon {
  font-size: 24px;
  width: 24px;
  height: 24px;
}
.nut-button-xlarge-children {
  font-size: 24px;
  font-weight: 600;
}
.nut-button-large {
  height: 40px;
  padding: 0px 16px;
  font-size: 16px;
  font-weight: 600;
}
.nut-button-large .nut-icon {
  font-size: 16px;
  width: 16px;
  height: 16px;
}
.nut-button-large-children {
  font-size: 16px;
  font-weight: 600;
}
.nut-button-small {
  height: 28px;
  padding: 0px 8px;
  font-size: 12px;
}
.nut-button-small .nut-icon {
  font-size: 12px;
  width: 12px;
  height: 12px;
}
.nut-button-small-children {
  font-size: 12px;
}
.nut-button-mini {
  height: 24px;
  padding: 0px 8px;
  font-size: 12px;
}
.nut-button-mini .nut-icon {
  font-size: 12px;
  width: 12px;
  height: 12px;
}
.nut-button-mini-children {
  font-size: 12px;
}
.nut-button-primary {
  color: #ffffff;
  background-origin: border-box;
  border-color: transparent;
}
.nut-button-primary-children {
  color: #ffffff;
}
.nut-button-primary-solid {
  background: #ff0f23;
  color: #ffffff;
  border-color: transparent;
}
.nut-button-primary-disabled {
  color: #ffffff;
  background: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-primary-solid-disabled {
  color: #ffffff;
  background: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-primary-none {
  color: #ff0f23;
}
.nut-button-primary-none-disabled {
  color: #ffadbe;
}
.nut-button-primary-outline {
  color: #ff0f23;
  border-color: #ff0f23;
}
.nut-button-primary-outline-disabled {
  color: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-primary-dashed {
  color: #ff0f23;
  border-color: #ff0f23;
}
.nut-button-primary-dashed-disabled {
  color: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-success {
  color: #ffffff;
  background: #2aa32a;
  background-origin: border-box;
  border-color: transparent;
}
.nut-button-success-children {
  color: #ffffff;
}
.nut-button-success-solid-disabled {
  background: #b2f0ae;
  border-color: #b2f0ae;
}
.nut-button-success-outline {
  color: #2aa32a;
  border-color: #2aa32a;
}
.nut-button-success-dashed {
  color: #2aa32a;
  border-color: #2aa32a;
}
.nut-button-success-outline-disabled {
  color: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-success-dashed-disabled {
  color: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-success-none {
  color: #2aa32a;
}
.nut-button-success-none-disabled {
  color: #b2f0ae;
}
.nut-button-info {
  color: #ffffff;
  background: #0073ff;
  background-origin: border-box;
  border-color: transparent;
}
.nut-button-info-children {
  color: #ffffff;
}
.nut-button-info-solid-disabled {
  background: #89a6f8;
  border-color: #89a6f8;
}
.nut-button-info-outline {
  color: #1988fa;
  border-color: #1988fa;
}
.nut-button-info-dashed {
  color: #1988fa;
  border-color: #1988fa;
}
.nut-button-info-outline-disabled {
  color: #89a6f8;
  border-color: #89a6f8;
}
.nut-button-info-dashed-disabled {
  color: #89a6f8;
  border-color: #89a6f8;
}
.nut-button-info-none {
  color: #1988fa;
}
.nut-button-info-none-disabled {
  color: #89a6f8;
}
.nut-button-danger {
  color: #ffffff;
  background: #ff0f23;
  background-origin: border-box;
  border-color: transparent;
}
.nut-button-danger-children {
  color: #ffffff;
}
.nut-button-danger-solid-disabled {
  background: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-danger-outline {
  color: #ff0f23;
  border-color: #ff0f23;
}
.nut-button-danger-dashed {
  color: #ff0f23;
  border-color: #ff0f23;
}
.nut-button-danger-outline-disabled {
  color: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-danger-dashed-disabled {
  color: #ffadbe;
  border-color: #ffadbe;
}
.nut-button-danger-none {
  color: #ff0f23;
}
.nut-button-danger-none-disabled {
  color: #ffadbe;
}
.nut-button-warning {
  color: #ffffff;
  background: #c47600;
  background-origin: border-box;
  border-color: transparent;
}
.nut-button-warning-children {
  color: #ffffff;
}
.nut-button-warning-disabled {
  color: #ffffff;
  background: #fdd3b9;
  border-color: #fdd3b9;
}
.nut-button-warning-solid-disabled {
  color: #ffffff;
  background: #fdd3b9;
  border-color: #fdd3b9;
}
.nut-button-warning-outline {
  color: #c47600;
  border-color: #c47600;
}
.nut-button-warning-dashed {
  color: #c47600;
  border-color: #c47600;
}
.nut-button-warning-outline-disabled {
  color: #fdd3b9;
  border-color: #fdd3b9;
}
.nut-button-warning-dashed-disabled {
  color: #fdd3b9;
  border-color: #fdd3b9;
}
.nut-button-warning-none {
  color: #c47600;
}
.nut-button-warning-none-disabled {
  color: #fdd3b9;
}
.nut-button-block {
  display: block;
  width: 100%;
}
.nut-button-outline {
  background: transparent;
  border-style: solid;
}
.nut-button-dashed {
  background: transparent;
  border-style: dashed;
}
.nut-button-none {
  background: transparent;
  border-color: transparent;
}
.nut-button-loading {
  cursor: default;
  opacity: 0.9;
}
.nut-button-round {
  border-radius: 8px;
}
.nut-button-round-xlarge {
  border-radius: 12px;
}
.nut-button-round-large {
  border-radius: 12px;
}
.nut-button-round-small {
  border-radius: 8px;
}
.nut-button-round-mini {
  border-radius: 6px;
}
.nut-button-square {
  border-radius: 0;
}

[dir=rtl] .nut-button-text {
  margin-left: 0;
  margin-right: 4px;
}

.nut-rtl .nut-button-text {
  margin-left: 0;
  margin-right: 4px;
}
[dir=rtl] .nut-button-text.right {
  margin-left: 4px;
}
.nut-rtl .nut-button-text.right {
  margin-left: 4px;
}
[dir=rtl] .nut-button::before,
.nut-rtl .nut-button::before {
  left: auto;
  right: 50%;
  transform: translate(50%, -50%);
}