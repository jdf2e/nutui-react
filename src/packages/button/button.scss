.nut-button {
  position: relative;
  display: flex;
  display: inline-block;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  height: $button-default-height;
  font-size: $button-default-font-size;
  font-weight: $button-default-font-weight;
  text-align: center;
  cursor: pointer;
  transition: $button-transition;
  -webkit-appearance: none;
  user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  color: $button-default-color;
  background: $button-default-background-color;

  &-text {
    margin-left: $button-text-icon-margin;

    &.right {
      margin-right: $button-text-icon-margin;
    }
  }

  &::before {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 100%;
    height: 100%;
    background-color: $color-mask;
    border: inherit;
    border-color: $color-mask;
    border-radius: inherit;
    transform: translate(-50%, -50%);
    opacity: 0;
    content: ' ';
  }

  &::after {
    border: none;
  }

  &:active::before {
    opacity: 0.1;
  }

  &-wrap {
    height: 100%;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: initial;
    .nut-icon {
      font-size: $button-default-font-size;
      width: $button-default-font-size;
      height: $button-default-font-size;
    }
  }

  &-loading,
  &-disabled {
    &::before {
      display: none;
    }
  }

  &.nut-button-icononly {
    width: $button-default-height;
    padding: 0;
  }

  &-default {
    padding: $button-default-padding;
    border: $button-border-width solid $button-default-border-color;

    &.nut-button-disabled {
      color: $button-default-disabled-color;
      background: transparent;
    }

    &.nut-button-none {
      background: transparent;

      &.nut-button-disabled {
        color: $button-default-disabled-color;
      }
    }

    &.nut-button-outline,
    &.nut-button-dashed {
      background: transparent;

      &.nut-button-disabled {
        color: $button-default-disabled-color;
        border-color: $button-default-disabled;
      }
    }
  }

  &-normal {
    padding: $button-normal-padding;
  }

  &-xlarge {
    height: $button-xlarge-height;
    padding: $button-xlarge-padding;
    font-size: $button-xlarge-font-size;
    font-weight: $button-large-font-weight;

    .nut-icon {
      font-size: $button-xlarge-font-size;
      width: $button-xlarge-font-size;
      height: $button-xlarge-font-size;
    }
  }

  &-large {
    height: $button-large-height;
    padding: $button-large-padding;
    font-size: $button-large-font-size;
    font-weight: $button-large-font-weight;

    .nut-icon {
      font-size: $button-large-font-size;
      width: $button-large-font-size;
      height: $button-large-font-size;
    }
  }

  &-small {
    height: $button-small-height;
    padding: $button-small-padding;
    font-size: $button-small-font-size;

    .nut-icon {
      font-size: $button-small-font-size;
      width: $button-small-font-size;
      height: $button-small-font-size;
    }
  }

  &-mini {
    height: $button-mini-height;
    padding: $button-mini-padding;
    font-size: $button-mini-font-size;

    .nut-icon {
      font-size: $button-mini-font-size;
      width: $button-mini-font-size;
      height: $button-mini-font-size;
    }
  }

  &-primary {
    color: $button-primary-color;
    background: $button-primary-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      background: $button-primary-disabled;
      border-color: $button-primary-disabled;
    }

    &.nut-button-none {
      background: transparent;
      color: $button-primary-border-color;

      &.nut-button-disabled {
        color: $button-primary-disabled;
      }
    }

    &.nut-button-outline {
      background: transparent;
      color: $button-primary-border-color;
      border: $button-border-width solid $button-primary-border-color;

      &.nut-button-disabled {
        color: $button-primary-disabled;
        border-color: $button-primary-disabled;
      }
    }

    &.nut-button-dashed {
      background: transparent;
      color: $button-primary-border-color;
      border: $button-border-width dashed $button-primary-border-color;

      &.nut-button-disabled {
        color: $button-primary-disabled;
        border-color: $button-primary-disabled;
      }
    }
  }

  &-success {
    color: $button-success-color;
    background: $button-success-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      background: $button-success-disabled;
      border-color: $button-success-disabled;
    }

    &.nut-button-outline,
    &.nut-button-dashed {
      color: $button-success-border-color;
      border-color: $button-success-border-color;
      background: transparent;

      &.nut-button-disabled {
        color: $button-primary-disabled;
        border-color: $button-primary-disabled;
      }
    }

    &.nut-button-none {
      color: $button-success-border-color;
    }
  }

  &-info {
    color: $button-info-color;
    background: $button-info-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      background: $button-info-disabled;
      border-color: $button-info-disabled;
    }

    &.nut-button-outline,
    &.nut-button-dashed {
      color: $button-info-border-color;
      border-color: $button-info-border-color;
      background: transparent;

      &.nut-button-disabled {
        color: $button-primary-disabled;
        border-color: $button-primary-disabled;
      }
    }

    &.nut-button-none {
      color: $button-info-border-color;
    }
  }

  &-danger {
    color: $button-danger-color;
    background: $button-danger-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      background: $button-danger-disabled;
      border-color: $button-danger-disabled;
    }

    &.nut-button-outline,
    &.nut-button-dashed {
      color: $button-danger-border-color;
      border-color: $button-danger-border-color;
      background: transparent;

      &.nut-button-disabled {
        color: $button-danger-disabled;
        border-color: $button-danger-disabled;
      }
    }

    &.nut-button-none {
      color: $button-danger-border-color;
    }
  }

  &-warning {
    color: $button-warning-color;
    background: $button-warning-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      background: $button-warning-disabled;
      border-color: $button-warning-disabled;
    }

    &.nut-button-outline,
    &.nut-button-dashed {
      color: $button-warning-border-color;
      border-color: $button-warning-border-color;
      background: transparent;

      &.nut-button-disabled {
        color: $button-warning-disabled;
        border-color: $button-warning-disabled;
      }
    }

    &.nut-button-none {
      color: $button-warning-border-color;
    }
  }

  &-block {
    display: block;
    width: 100%;
  }

  &-solid {
    color: $button-primary-color;
    background: $button-primary-background-color;
    background-origin: border-box;
    border: $button-border-width solid transparent;

    &.nut-button-disabled {
      color: $button-default-disabled-color;
      background: $button-default-disabled;
    }
  }

  &.nut-button-outline,
  &.nut-button-dashed {
    background: transparent;
  }

  &.nut-button-none {
    background: transparent;
    border-color: transparent;
    background: transparent;
  }

  &-disabled {
    cursor: not-allowed;
    color: #ffffff;
    background: $button-default-disabled;
    border-color: $button-default-disabled;
  }

  &-loading {
    cursor: default;
    opacity: 0.9;
  }

  &-round {
    border-radius: $button-border-radius;

    &.nut-button-xlarge,
    &.nut-button-large {
      border-radius: $button-large-border-radius;
    }

    &.nut-button-small {
      border-radius: $button-small-border-radius;
    }

    &.nut-button-mini {
      border-radius: $button-mini-border-radius;
    }
  }

  &-square {
    border-radius: $button-square-border-radius;
  }
}

[dir='rtl'] .nut-button,
.nut-rtl .nut-button {
  &-text {
    margin-left: 0;
    margin-right: $button-text-icon-margin;

    &.right {
      margin-left: $button-text-icon-margin;
    }
  }
  &::before {
    left: auto;
    right: 50%;
    transform: translate(50%, -50%);
  }
}
