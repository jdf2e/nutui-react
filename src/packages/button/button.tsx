import React, { CSSProperties, useCallback } from 'react'
import type { MouseEvent } from 'react'
import classNames from 'classnames'
import { Loading } from '@nutui/icons-react'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export type ButtonType =
  | 'default'
  | 'primary'
  | 'info'
  | 'success'
  | 'warning'
  | 'danger'
export type ButtonSize = 'xlarge' | 'large' | 'normal' | 'small' | 'mini'
export type ButtonShape = 'square' | 'round'
export type ButtonFill = 'solid' | 'outline' | 'dashed' | 'none'

export interface ButtonProps extends BasicComponent {
  color: string
  shape: ButtonShape
  type: ButtonType
  size: ButtonSize
  fill: ButtonFill
  block: boolean
  loading: boolean
  disabled: boolean
  icon: React.ReactNode
  rightIcon: React.ReactNode
  id: string
  nativeType: 'submit' | 'reset' | 'button'
  onClick: (e: MouseEvent<HTMLButtonElement>) => void
}

const prefixCls = 'nut-button'

const defaultProps = {
  ...ComponentDefaults,
  color: '',
  type: 'default',
  size: 'normal',
  shape: 'round',
  fill: 'outline',
  loading: false,
  disabled: false,
  block: false,
  icon: null,
  rightIcon: null,
  nativeType: 'button',
  onClick: (e: MouseEvent<HTMLButtonElement>) => {},
} as ButtonProps
export const Button = React.forwardRef<HTMLButtonElement, Partial<ButtonProps>>(
  (props, ref) => {
    const {
      color,
      shape,
      fill,
      loading,
      disabled,
      type,
      size,
      block,
      icon,
      rightIcon,
      children,
      className,
      style,
      nativeType,
      onClick,
      ...rest
    } = {
      ...defaultProps,
      ...props,
    }
    const getStyle = useCallback(() => {
      const style: CSSProperties = {}
      if (props.color) {
        if (props.fill === 'outline' || props.fill === 'dashed') {
          style.color = color
          if (!color?.includes('gradient')) {
            style.borderColor = color
          }
        } else {
          style.color = '#fff'
          style.background = color
          style.borderColor = 'transparent'
        }
      }
      return style
    }, [color])

    const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
      if (!loading && !disabled && onClick) {
        onClick(e)
      }
    }

    return (
      <button
        {...rest}
        ref={ref}
        type={nativeType}
        className={classNames(
          prefixCls,
          `${prefixCls}-${type}`,
          props.fill ? `${prefixCls}-${fill}` : null,
          children ? '' : `${prefixCls}-icononly`,
          {
            [`${prefixCls}-${size}`]: size,
            [`${prefixCls}-${shape}`]: shape,
            [`${prefixCls}-block`]: block,
            [`${prefixCls}-disabled`]: disabled || loading,
            [`${prefixCls}-loading`]: loading,
          },
          className
        )}
        style={{ ...getStyle(), ...style }}
        onClick={(e) => handleClick(e)}
      >
        <div className="nut-button-wrap">
          {loading && <Loading className="nut-icon-loading" />}
          {!loading && icon ? icon : null}
          {children && (
            <div
              className={`${icon || loading ? 'nut-button-text' : ''}${
                rightIcon ? ' nut-button-text right' : ''
              }`}
            >
              {children}
            </div>
          )}
          {rightIcon || null}
        </div>
      </button>
    )
  }
)

Button.displayName = 'NutButton'
