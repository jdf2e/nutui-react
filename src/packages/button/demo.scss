.button-cell {
  flex-wrap: wrap;
}
