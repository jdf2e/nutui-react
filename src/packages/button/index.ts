import { Button } from './button'

export type {
  ButtonType,
  ButtonProps,
  ButtonSize,
  ButtonShape,
  ButtonFill,
} from './button'
export default Button
