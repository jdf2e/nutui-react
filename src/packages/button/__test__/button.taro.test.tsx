import TestUtils from '@tarojs/test-utils-react'
import '@testing-library/jest-dom'
import { Button } from '../button.taro'

const testUtils = new TestUtils()
const execFn = (desc: string, component: any, props: object, cb: any) => {
  it(desc, async () => {
    await testUtils.mount(component, {
      props,
    })
    cb()
    testUtils.unmout()
  })
}
describe('button-taro', () => {
  execFn(
    'should match snapshot',
    Button,
    {
      type: 'primary',
      shape: 'round',
      children: '主要按钮',
    },
    () => {
      expect(testUtils.html()).toMatchSnapshot()
    }
  )
  execFn(
    'should fill is outline',
    Button,
    {
      type: 'primary',
      shape: 'round',
      fill: 'outline',
      children: '主要按钮',
    },
    () => {
      expect(testUtils.queries.querySelector('.nut-button')).toHaveClass(
        'nut-button-outline'
      )
    }
  )
  execFn(
    'should props color when use fill correctly',
    Button,
    {
      type: 'primary',
      fill: 'outline',
      children: '主要按钮',
      color: 'blue',
    },
    () => {
      expect(testUtils.queries.querySelector('.nut-button')).toHaveStyle({
        'border-color': 'blue',
      })
    }
  )
  execFn(
    'should props color with no fill correctly',
    Button,
    {
      type: 'primary',
      children: '主要按钮',
      color: 'blue',
    },
    () => {
      expect(testUtils.queries.querySelector('.nut-button')).toHaveStyle({
        background: 'blue',
      })
    }
  )
  execFn(
    'should children correctly',
    Button,
    {
      type: 'primary',
      children: '主要按钮',
      color: 'blue',
    },
    () => {
      expect(testUtils.queries.querySelector('.nut-button')).toHaveTextContent(
        '主要按钮'
      )
    }
  )
  execFn(
    'should show loading correctly',
    Button,
    {
      type: 'primary',
      children: '主要按钮',
      color: 'blue',
      loading: true,
    },
    () => {
      expect(testUtils.queries.querySelector('.nut-button')).toHaveClass(
        'nut-button-loading'
      )
    }
  )
})
