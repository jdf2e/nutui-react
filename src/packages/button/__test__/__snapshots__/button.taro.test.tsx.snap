// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button-taro should match snapshot 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<button[39m
      [33mclass[39m=[32m"nut-button nut-button-primary nut-button-normal nut-button-round"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"nut-button-wrap"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m""[39m
        [36m>[39m
          [0m‰∏ªË¶ÅÊåâÈíÆ[0m
        [36m</div>[39m
      [36m</div>[39m
    [36m</button>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
